pennylane/fermi/fermionic.py

```diff
@@ -160,3 +160,81 @@ def __pow__(self, value):
 # TODO: create __sub__ method when PauliSentence is merged.
 # TODO: support multiply by number in __mul__ when PauliSentence is merged.
 # TODO: create mapping method when the ne jordan_wigner function is added.
+
+
+class FermiSentence(dict):
+    r"""Immutable dictionary used to represent a Fermi sentence, a linear combination of Fermi words, with the keys
+    as FermiWord instances and the values correspond to coefficients.
+
+    >>> w1 = FermiWord({(0, 0) : '+', (1, 1) : '-'})
+    >>> w2 = FermiWord({(0, 1) : '+', (1, 2) : '-'})
+    >>> s = FermiSentence({w1 : 1.2, w2: 3.1})
+    >>> s
+    1.2 * '0+ 1-'
+    + 3.1 * '1+ 2-'
+    """
+
+    @property
+    def wires(self):
+        r"""Return wires of the FermiSentence."""
+        return set().union(*(fw.wires for fw in self.keys()))
+
+    def __str__(self):
+        r"""String representation of a FermiSentence."""
+        if len(self) == 0:
+            return "0 * ''"
+        return "\n+ ".join(f"{coeff} * '{fw.to_string()}'" for fw, coeff in self.items())
+
+    def __repr__(self):
+        r"""Terminal representation for FermiSentence."""
+        return str(self)
+
+    def __missing__(self, key):
+        r"""If the FermiSentence does not contain a FermiWord then the associated value will be 0."""
+        return 0.0
+
+    def __add__(self, other):
+        r"""Add two Fermi sentence together by iterating over the smaller one and adding its terms
+        to the larger one."""
+        smaller_fs, larger_fs = (
+            (self, copy(other)) if len(self) < len(other) else (other, copy(self))
+        )
+        for key in smaller_fs:
+            larger_fs[key] += smaller_fs[key]
+
+        return larger_fs
+
+    def __mul__(self, other):
+        r"""Multiply two Fermi sentences by iterating over each sentence and multiplying the Fermi
+        words pair-wise"""
+
```

Comment:
We should allow multiplying a `FermiSentence` by a `FermiWord` with something like:
```
if isinstance(other, FermiWord):
    other = FermiSentence({other: 1.0})
```

Similarly, multiplying a `FermiWord`  by a `FermiSentence` should be possible.